lightning-invoice/src/payment.rs

```diff
@@ -466,6 +494,46 @@ where
 		}.map_err(|e| PaymentError::Sending(e))
 	}
 
+	fn process_path_inflight_htlcs(&self, hops: &Vec<RouteHop>) {
```

Comment:
I think the PR would benefit from a better description of what exactly is aimed. AFAICT, it introduces tracking of in-flight HTLCs issued by the sender to update in consequence scoring's `DirectedChannelLiquidity` (to be introduced in future commit IIUC the direction) and generate more accurate scores ?

One interesting follow-up could be to track the CLTV delta as well, to know either when all the HTLC are going to expire and the link is going to be ready again or there is a HTLC stuck and the link should be disregarded (okay we might have few more rounds of iterations before we do that kind of scoring optimizations).

Reply:
@ariard I updated the PR description to reflect the current discussion of what we are doing :)

Reply:
Thanks for that!